src/ck-core/ck.C

```diff
@@ -1717,7 +1721,12 @@ static inline void _sendMsgBranch(int eIdx, void *msg, CkGroupID gID,
         env = _prepareImmediateMsgBranch(eIdx,msg,gID,ForBocMsg);
     }else
     {
-        env = _prepareMsgBranch(eIdx,msg,gID,ForBocMsg);
+#if CMK_SMP
```

Comment:
Adding the `CMK_SMP` guard here is causing the hang because bcast messages for UTH are now marked as `ForBocMsg` instead of `BocBcastMsg`. And it seems like UTH uses some of the within-node code path. 

I think I can get over the hang by changing the guard to `CMK_SMP && CMK_ONESIDED`, since it's primarily required for the Post API (since it uses the same message twice) and UTH has CMK_ONESIDED set to 0. However, I'm curious as to why UTH requires the bcast message to be `BocBcastMsg`.

Reply:
Would that change affect non-SMP netlrts, or multicore?

Reply:
I don't think it's strictly required. But I think it was done that way to minimize the number of different code paths. I also don't think it was treated specially during the initial implementation. I don't think I did any UTH specific testing or changes.

Reply:
And in theory I don't think the change Nitin proposed should adversely affect the layers you mentioned. Initially everything worked without the SMP guard except ZC. They all followed the same exact codepath to forward broadcasts to the rest of the node (whatever a node was defined as for those layers).

Reply:
Oh okay, thanks. I'll try finding out the code path used by the `Uth` in master to determine what it's doing differently. (as compared to other layers). 
My suggested solution will require a guard for `Uth` specifically because I realized that I had added the `CMK_SMP` guard for all nonSMP builds (on CMK_ONESIDED_IMPL layers) and `Uth` is a not `CMK_SMP` but seems to behave like an SMP layer.